internal/rbd/driver/driver.go

```diff
@@ -125,25 +125,21 @@ func (r *Driver) Run(conf *util.Config) {
 			})
 	}
 
-	if k8s.RunsOnKubernetes() {
-		nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
-		if err != nil {
-			log.FatalLogMsg(err.Error())
-		}
-	}
-
-	if conf.EnableReadAffinity {
-		crushLocationMap = util.GetCrushLocationMap(conf.CrushLocationLabels, nodeLabels)
-	}
-
 	// Create GRPC servers
 	r.ids = NewIdentityServer(r.cd)
 
-	if conf.IsNodeServer {
+	if conf.IsNodeServer && k8s.RunsOnKubernetes() {
 		topology, err = util.GetTopologyFromDomainLabels(conf.DomainLabels, conf.NodeID, conf.DriverName)
 		if err != nil {
 			log.FatalLogMsg(err.Error())
 		}
+		nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
+		if err != nil {
+			log.FatalLogMsg(err.Error())
+		}
+		if conf.EnableReadAffinity {
+			crushLocationMap = util.GetCrushLocationMap(conf.CrushLocationLabels, nodeLabels)
+		}
```

Comment:
```suggestion
	if conf.IsNodeServer {
		topology, err = util.GetTopologyFromDomainLabels(conf.DomainLabels, conf.NodeID, conf.DriverName)
		if err != nil {
			log.FatalLogMsg(err.Error())
		}
		if k8s.RunsOnKubernetes() {
		        nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
		        if err != nil {
			        log.FatalLogMsg(err.Error())
		        }
		}
		if conf.EnableReadAffinity {
			crushLocationMap = util.GetCrushLocationMap(conf.CrushLocationLabels, nodeLabels)
		}
```

The current change will prevent nodeserver from running in non-K8s environment.
Let's check runsOnK8s only while getting node labels.

Reply:
@Rakshith-R Actually, this is tricky, because `util.GetTopologyFromDomainLabels` internally also fetches node labels using `k8s.GetNodeLabels` that's why I merged the condition. I suggest to extend `util.GetTopologyFromDomainLabels` by new parameter nodeLabels so we can separately check if the controller is running in k8s and fetch the node labels and then just reuse them

Reply:
> @Rakshith-R Actually, this is tricky, because `util.GetTopologyFromDomainLabels` internally also fetches node labels using `k8s.GetNodeLabels` that's why I merged the condition. I suggest to extend `util.GetTopologyFromDomainLabels` by new parameter nodeLabels so we can separately check if the controller is running in k8s and fetch the node labels and then just reuse them

@nemcikjan 
We'll now be having nodes get rbac by default for nodeserver so let's just make sure GetNodeLabels is covered inside conf.nodeserver struct for backport.

We can make this enhancement in follow-up pr.

Reply:
@Rakshith-R OK, I'll make appropriate changes to the scope of this PR

Reply:
> Actually, this is tricky, because `util.GetTopologyFromDomainLabels` internally also fetches node labels using `k8s.GetNodeLabels` that's why I merged the condition. I suggest to extend `util.GetTopologyFromDomainLabels` by new parameter nodeLabels so we can separately check if the controller is running in k8s and fetch the node labels and then just reuse them

That part should be fine since we are now adding rbac always for nodeserver.
let's just make this change to keep GetNodeLabels inside conf.nodeserver  for backports.
We'll make that enhancement in a follow up pr. 

Please check the commitlint failure too 
https://github.com/ceph/ceph-csi/actions/runs/7126377653/job/19405449656?pr=4302

Thanks

Reply:
@nemcikjan ^